File: lib/spoonacular_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class SpoonacularService {
  final String apiKey = '47479824a9a442f486fb2cd1059d51c0'; // Replace with your key
  final String baseUrl = 'https://api.spoonacular.com';

  Future<Map<String, dynamic>?> searchRecipes(String query) async {
    final url = Uri.parse('$baseUrl/recipes/complexSearch?query=$query&apiKey=$apiKey');

    try {
      final response = await http.get(url);

      if (response.statusCode == 200) {
        return json.decode(response.body);
      } else {
        print('Failed to load recipes: ${response.statusCode}');
        return null;
      }
    } catch (e) {
      print('Error: $e');
      return null;
    }
  }

  Future<Map<String, dynamic>?> searchRecipesPaginated(String query, int offset, int limit) async {
    try {
      final url = Uri.parse('$baseUrl/recipes/complexSearch?query=$query&offset=$offset&number=$limit&apiKey=$apiKey');
      final response = await http.get(url);

      if (response.statusCode == 200) {
        return json.decode(response.body);
      } else {
        print('Failed to load recipes: ${response.statusCode}');
      }
    } catch (e) {
      print('Error: $e');
    }
    return null;
  }
}

File: src/services/spoonacularService.js
// src/services/spoonacularService.js

const API_KEY = "0e2a083a8ead436e883e2e9f3f135f83";
const BASE_URL = "https://api.spoonacular.com/recipes/complexSearch";

/**
 * Fetch recipes from Spoonacular with optional filters
 */
export const fetchRecipesFromSpoonacular = async ({
  query,
  diet = "",
  cuisine = "",
  type = "",
  maxResults = 10,
}) => {
  const url = new URL(BASE_URL);
  url.searchParams.set("apiKey", API_KEY);
  url.searchParams.set("query", query);
  url.searchParams.set("addRecipeInformation", "true");
  url.searchParams.set("number", maxResults);

  if (diet) url.searchParams.set("diet", diet);
  if (cuisine) url.searchParams.set("cuisine", cuisine);
  if (type) url.searchParams.set("type", type);

  try {
    const response = await fetch(url);
    const data = await response.json();

    if (!response.ok) {
      throw new Error(data.message || "Spoonacular API fetch failed");
    }

    return data.results || [];
  } catch (err) {
    console.error("❌ Spoonacular fetch error:", err.message);
    return [];
  }
};
